### PR TITLE
python3Packages.anyio: 3.2.0 -> 3.2.1

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

https://github.com/agronholm/anyio/blob/3.2.1/docs/versionhistory.rst
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 128902` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages failed to build:</summary>
  <ul>
    <li>dyndnsc</li>
    <li>home-assistant</li>
    <li>lexicon</li>
    <li>python38Packages.authcaptureproxy</li>
    <li>python38Packages.dependency-injector</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.ipfshttpclient</li>
    <li>python38Packages.json-logging</li>
    <li>python38Packages.teslajsonpy</li>
    <li>python39Packages.authcaptureproxy</li>
    <li>python39Packages.dependency-injector</li>
    <li>python39Packages.fastapi</li>
    <li>python39Packages.ipfshttpclient</li>
    <li>python39Packages.json-logging</li>
    <li>python39Packages.teslajsonpy</li>
  </ul>
</details>
<details>
  <summary>60 packages built:</summary>
  <ul>
    <li>datasette (python38Packages.datasette)</li>
    <li>python38Packages.PyRMVtransport</li>
    <li>python38Packages.anyio</li>
    <li>python38Packages.axis</li>
    <li>python38Packages.denonavr</li>
    <li>python38Packages.elmax</li>
    <li>python38Packages.entrance</li>
    <li>python38Packages.entrance-with-router-features</li>
    <li>python38Packages.envoy-reader</li>
    <li>python38Packages.ha-philipsjs</li>
    <li>python38Packages.httpcore</li>
    <li>python38Packages.httpx</li>
    <li>python38Packages.ircrobots</li>
    <li>python38Packages.ismartgate</li>
    <li>python38Packages.jupyter_server</li>
    <li>python38Packages.jupyterlab</li>
    <li>python38Packages.jupyterlab_server</li>
    <li>python38Packages.jupytext</li>
    <li>python38Packages.nbclassic</li>
    <li>python38Packages.pylitterbot</li>
    <li>python38Packages.pypoint</li>
    <li>python38Packages.pytest-httpx</li>
    <li>python38Packages.pytest-sanic</li>
    <li>python38Packages.respx</li>
    <li>python38Packages.sanic</li>
    <li>python38Packages.sanic-auth</li>
    <li>python38Packages.sanic-testing</li>
    <li>python38Packages.sunpy</li>
    <li>python38Packages.total-connect-client</li>
    <li>python38Packages.zeep</li>
    <li>python39Packages.PyRMVtransport</li>
    <li>python39Packages.anyio</li>
    <li>python39Packages.axis</li>
    <li>python39Packages.datasette</li>
    <li>python39Packages.denonavr</li>
    <li>python39Packages.elmax</li>
    <li>python39Packages.entrance</li>
    <li>python39Packages.entrance-with-router-features</li>
    <li>python39Packages.envoy-reader</li>
    <li>python39Packages.ha-philipsjs</li>
    <li>python39Packages.httpcore</li>
    <li>python39Packages.httpx</li>
    <li>python39Packages.ircrobots</li>
    <li>python39Packages.ismartgate</li>
    <li>python39Packages.jupyter_server</li>
    <li>python39Packages.jupyterlab</li>
    <li>python39Packages.jupyterlab_server</li>
    <li>python39Packages.jupytext</li>
    <li>python39Packages.nbclassic</li>
    <li>python39Packages.pylitterbot</li>
    <li>python39Packages.pypoint</li>
    <li>python39Packages.pytest-httpx</li>
    <li>python39Packages.pytest-sanic</li>
    <li>python39Packages.respx</li>
    <li>python39Packages.sanic</li>
    <li>python39Packages.sanic-auth</li>
    <li>python39Packages.sanic-testing</li>
    <li>python39Packages.sunpy</li>
    <li>python39Packages.total-connect-client</li>
    <li>python39Packages.zeep</li>
  </ul>
</details>

**Comment by siraben** [comment]
Were the failures already here before?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 128902` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>14 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>dyndnsc: log was empty</li>
<li>lexicon: log was empty</li>
<li>python38Packages.authcaptureproxy: log was empty</li>
<li>python38Packages.dependency-injector: log was empty</li>
<li>python38Packages.fastapi: log was empty</li>
<li>python38Packages.ipfshttpclient: log was empty</li>
<li>python38Packages.json-logging: log was empty</li>
<li>python38Packages.teslajsonpy: log was empty</li>
<li>python39Packages.authcaptureproxy: log was empty</li>
<li>python39Packages.dependency-injector: log was empty</li>
<li>python39Packages.fastapi: log was empty</li>
<li>python39Packages.ipfshttpclient: log was empty</li>
<li>python39Packages.json-logging: log was empty</li>
<li>python39Packages.teslajsonpy: log was empty</li>
  </ul>
</details>
<details>
<summary>
61 packages built:
</summary>
<ul>
<li>datasette (python38Packages.datasette)</li>
<li>home-assistant</li>
<li>python38Packages.PyRMVtransport</li>
<li>python38Packages.anyio</li>
<li>python38Packages.axis</li>
<li>python38Packages.denonavr</li>
<li>python38Packages.elmax</li>
<li>python38Packages.entrance</li>
<li>python38Packages.entrance-with-router-features</li>
<li>python38Packages.envoy-reader</li>
<li>python38Packages.ha-philipsjs</li>
<li>python38Packages.httpcore</li>
<li>python38Packages.httpx</li>
<li>python38Packages.ircrobots</li>
<li>python38Packages.ismartgate</li>
<li>python38Packages.jupyter_server</li>
<li>python38Packages.jupyterlab</li>
<li>python38Packages.jupyterlab_server</li>
<li>python38Packages.jupytext</li>
<li>python38Packages.nbclassic</li>
<li>python38Packages.pylitterbot</li>
<li>python38Packages.pypoint</li>
<li>python38Packages.pytest-httpx</li>
<li>python38Packages.pytest-sanic</li>
<li>python38Packages.respx</li>
<li>python38Packages.sanic</li>
<li>python38Packages.sanic-auth</li>
<li>python38Packages.sanic-testing</li>
<li>python38Packages.sunpy</li>
<li>python38Packages.total-connect-client</li>
<li>python38Packages.zeep</li>
<li>python39Packages.PyRMVtransport</li>
<li>python39Packages.anyio</li>
<li>python39Packages.axis</li>
<li>python39Packages.datasette</li>
<li>python39Packages.denonavr</li>
<li>python39Packages.elmax</li>
<li>python39Packages.entrance</li>
<li>python39Packages.entrance-with-router-features</li>
<li>python39Packages.envoy-reader</li>
<li>python39Packages.ha-philipsjs</li>
<li>python39Packages.httpcore</li>
<li>python39Packages.httpx</li>
<li>python39Packages.ircrobots</li>
<li>python39Packages.ismartgate</li>
<li>python39Packages.jupyter_server</li>
<li>python39Packages.jupyterlab</li>
<li>python39Packages.jupyterlab_server</li>
<li>python39Packages.jupytext</li>
<li>python39Packages.nbclassic</li>
<li>python39Packages.pylitterbot</li>
<li>python39Packages.pypoint</li>
<li>python39Packages.pytest-httpx</li>
<li>python39Packages.pytest-sanic</li>
<li>python39Packages.respx</li>
<li>python39Packages.sanic</li>
<li>python39Packages.sanic-auth</li>
<li>python39Packages.sanic-testing</li>
<li>python39Packages.sunpy</li>
<li>python39Packages.total-connect-client</li>
<li>python39Packages.zeep</li>
</ul>
</details>
